python/lsst/obs/decam/isr.py

```diff
@@ -53,17 +57,55 @@ class DecamIsrTask(IsrTask):
     ConfigClass = DecamIsrConfig
 
     def convertIntToFloat(self, exp):
-        """No conversion necessary."""
+        """No conversion necessary.
+        """
         return exp
 
+    def readIsrData(self, dataRef, exposure):
+        """Retrieve necessary data products for instrument signature removal.
+
+        This is identical to ip.isr.isrTask.readIsrData, except it also
+        includes crosstalkSources in the isrData Struct that is returned.
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.butlerSubset.ButlerDataRef`
+            dataRef corresponding to the exposure to process
+        exposure : `lsst.afw.image.Exposure`
+            exposure to process
+
+        Returns
+        -------
+        `lsst.pipe.base.Struct`
+        Struct with components:
+        - isrData : includes all the same components returned by ip_isr's IsrTask.readIsrData
+        - crosstalkSources : `defaultdict`
+                                 included if crosstalk correction is turned on
+        """
+        isrData = super().readIsrData(dataRef, exposure)
+        if self.config.doCrosstalk:
+            crosstalkSources = self.crosstalk.prepCrosstalk(dataRef)
```

Comment:
I'd like to see this in the base class `IsrTask` so that other instruments can use this mechanism without reinventing wheels.